R/utils.R

```diff
@@ -1187,12 +1187,13 @@ split_big_table_for_xlsx <- function(dt_list,
       if (to_big_data_list[[i]][1] && to_big_data_list[[i]][2]) {
         stop("the array is too large in both dimensions, run the functions one dimension at a time")
       } else if (to_big_data_list[[i]][1]) {
-        out_list[length(out_list) + 1] <- list(dt_list[[i]][seq_len(max_row)])
+        # using seq_len here causes the output format to change, e.g. from data.table to integer
+        out_list[length(out_list) + 1] <- list(dt_list[[i]][1:max_row])
```

Comment:
```suggestion
        out_list[length(out_list) + 1] <- list(dt_list[[i]][c(seq_len(max_row)), ])
```

Bioconductor linter does not like `1:...`